### ISSUE TITLE
Web interface returns forbidden and cannot be accessed

### ISSUE THREAD

**Issue by taku0220** [owntone/owntone-server#1239]
I am not very good at English, so I apologize in advance.

I am testing V27.4 on a Synology NAS DS220j.
I have cross-compiled forked-daapd for this NAS using the [SynoCommunity/spksrc](https://github.com/SynoCommunity/spksrc) toolchain.

If the path of webroot_directory contains symlink, accessing the web interface will return forbidden.

```
[DEBUG]    httpd: Starting web server with root directory '/var/packages/forked-daapd/target/share/forked-daapd/htdocs'
...
[ WARN]    httpd: Access to file outside the web root dir forbidden: /volume1/@appstore/forked-daapd/share/forked-daapd/htdocs/index.html
```

'/var/packages/forked-daapd/target' is a symbolic link for '/volume1/@appstore/forked-daapd'.
```
# file /var/packages/forked-daapd/target
/var/packages/forked-daapd/target: symbolic link to /volume1/@appstore/forked-daapd
```

I think it's a problem with serve_file(), but I don't have the ability to fix it safely, so I'll just report it.

I've been able to work around this using the -w option.
```
cd $(readlink -f /var/packages/forked-daapd/target)
WEBDIR=$(readlink -f /var/packages/forked-daapd/target/share/forked-daapd/htdocs)
sbin/forked-daapd -f -c etc/forked-daapd.conf -w ${WEBDIR}
```

**Comment by ejurgensen** [comment]
Thanks for the report (and the language is fine!). @chme I guess this is for you to consider.

@taku0220 I think you can also specify an alternate path with ./configure by using either the datarootdir or the datadir setting. As far as I can tell, forked-daapd will look for the files in [DATADIR]/htdocs

**Comment by smammy** [comment]
I'm running into this as well since I use GNU Stow to manage locally-installed software. OwnTone was configured with `./configure` (using the default prefix of `/usr/local`) and installed with `make install prefix=/usr/local/stow/owntone`. `/usr/local/share/owntone` is a symlink to `../stow/owntone/share/owntone`.

I understand that symlinks can cause security issues (or information disclosure, at least). I'm wondering if it would be acceptable for owntone to allow symlinks to exist in the path _to_ the webroot, but not _within_ the webroot. That is, my scenario and @taku0220's scenario would be fine, but if, say, `htdocs/player/js` was a symlink pointing somewhere wild would still be rejected.

In the mean time, I'm using `-w /usr/local/stow/owntone/share/owntone/htdocs` in `owntone.service` to work around this.

**Comment by whatdoineed2do** [comment]
@taku0220 - in `httpd.c`, it performs a `path_is_legal()` that takes a dereferenced (via `realpath()`) requested path against the webroot directory which has not be deferenced.  A simple fix in `httpd.c` to this could be:
```
//static const *char webroot_directory;
static char webroot_directory[PATH_MAX] = { 0 };
....
int
httpd_init(...)
{
...
  //webroot_directory = webroot;
  realpath(webroot, webroot_directory);
...
```

**Comment by chme** [comment]
@whatdoineed2do thanks for the analysis and the fix! I opened a PR for this #1282
